vtkext/private/module/vtkF3DImguiObserver.cxx

```diff
@@ -0,0 +1,291 @@
+    if (keySym == "t" || keySym == "T") return ImGuiKey_T;
+    if (keySym == "u" || keySym == "U") return ImGuiKey_U;
+    if (keySym == "v" || keySym == "V") return ImGuiKey_V;
+    if (keySym == "w" || keySym == "W") return ImGuiKey_W;
+    if (keySym == "x" || keySym == "X") return ImGuiKey_X;
+    if (keySym == "y" || keySym == "Y") return ImGuiKey_Y;
+    if (keySym == "z" || keySym == "Z") return ImGuiKey_Z;
+    if (keySym == "F1") return ImGuiKey_F1;
+    if (keySym == "F2") return ImGuiKey_F2;
+    if (keySym == "F3") return ImGuiKey_F3;
+    if (keySym == "F4") return ImGuiKey_F4;
+    if (keySym == "F5") return ImGuiKey_F5;
+    if (keySym == "F6") return ImGuiKey_F6;
+    if (keySym == "F7") return ImGuiKey_F7;
+    if (keySym == "F8") return ImGuiKey_F8;
+    if (keySym == "F9") return ImGuiKey_F9;
+    if (keySym == "F10") return ImGuiKey_F10;
+    if (keySym == "F11") return ImGuiKey_F11;
+    if (keySym == "F12") return ImGuiKey_F12;
+    if (keySym == "F13") return ImGuiKey_F13;
+    if (keySym == "F14") return ImGuiKey_F14;
+    if (keySym == "F15") return ImGuiKey_F15;
+    if (keySym == "F16") return ImGuiKey_F16;
+    if (keySym == "F17") return ImGuiKey_F17;
+    if (keySym == "F18") return ImGuiKey_F18;
+    if (keySym == "F19") return ImGuiKey_F19;
+    if (keySym == "F20") return ImGuiKey_F20;
+    if (keySym == "F21") return ImGuiKey_F21;
+    if (keySym == "F22") return ImGuiKey_F22;
+    if (keySym == "F23") return ImGuiKey_F23;
+    if (keySym == "F24") return ImGuiKey_F24;
+    if (keySym == "apostrophe") return ImGuiKey_Apostrophe;
+    if (keySym == "comma") return ImGuiKey_Comma;
+    if (keySym == "minus") return ImGuiKey_Minus;
+    if (keySym == "period") return ImGuiKey_Period;
+    if (keySym == "slash") return ImGuiKey_Slash;
+    if (keySym == "semicolon") return ImGuiKey_Semicolon;
+    if (keySym == "equal") return ImGuiKey_Equal;
+    if (keySym == "bracketleft") return ImGuiKey_LeftBracket;
+    if (keySym == "backslash") return ImGuiKey_Backslash;
+    if (keySym == "bracketright") return ImGuiKey_RightBracket;
+    if (keySym == "grave") return ImGuiKey_GraveAccent;
+    if (keySym == "Caps_Lock") return ImGuiKey_CapsLock;
+    if (keySym == "Scroll_Lock") return ImGuiKey_ScrollLock;
+    if (keySym == "Num_Lock") return ImGuiKey_NumLock;
+    if (keySym == "Snapshot") return ImGuiKey_PrintScreen;
+    if (keySym == "Pause") return ImGuiKey_Pause;
+    if (keySym == "KP_0") return ImGuiKey_Keypad0;
+    if (keySym == "KP_1") return ImGuiKey_Keypad1;
+    if (keySym == "KP_2") return ImGuiKey_Keypad2;
+    if (keySym == "KP_3") return ImGuiKey_Keypad3;
+    if (keySym == "KP_4") return ImGuiKey_Keypad4;
+    if (keySym == "KP_5") return ImGuiKey_Keypad5;
+    if (keySym == "KP_6") return ImGuiKey_Keypad6;
+    if (keySym == "KP_7") return ImGuiKey_Keypad7;
+    if (keySym == "KP_8") return ImGuiKey_Keypad8;
+    if (keySym == "KP_9") return ImGuiKey_Keypad9;
```

Comment:
Would it be worth it to consider a log(n) implementation of this ?

Reply:
I made it O(1)

Reply:
right, i always forgot about unordered map ^^